contribs/gnogenesis/integration_test.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/tm2/pkg/bft/rpc/client"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/ed25519"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// This integration test try to ensure `gnogenesis` is up to date against gno.land, and
+// the generated `genesis.json` is valid and can be loaded by the node.
+func TestIntegration(t *testing.T) {
+	const (
+		chainid     = "test"
+		genesisTime = "1750402800" // Friday, June 20th 2025 09:00 GMT+0200 (Central European Summer Time)
+	)
+
+	ctx := context.Background()
+
+	logger := log.NewTestingLogger(t)
+	gnoroot := gnoenv.RootDir()
+
+	genesisfile := filepath.Join(t.TempDir(), "genesis.json")
+
+	// Generate file
+	runGnoGenesisCommand(t, ctx, "generate",
+		"-chain-id", chainid,
+		"-genesis-time", genesisTime,
+		"-output-path", genesisfile,
+	)
+
+	// Create keybase
+	gnoHomeDir := filepath.Join(t.TempDir(), "gno")
+
+	pk := ed25519.GenPrivKey()
+	myVal := bft.NewMockPVWithParams(pk, false, false)
+	kb, err := keys.NewKeyBaseFromDir(gnoHomeDir)
+	require.NoError(t, err)
+	kb.ImportPrivKey("mykey", pk, "")
+
+	// Add my validator
+	runGnoGenesisCommand(t, ctx, "validator", "add",
+		"--name", "myval",
+		"--genesis-path", genesisfile,
+		"--address", myVal.GetPubKey().Address().String(),
+		"--pub-key", myVal.GetPubKey().String(),
+		"--power", "1",
+	)
+
+	// Dummy account
+	dKeys := common.GetDummyKeys(t, 3)
+
+	// Generate balance sheet
+	defaultBalanceAmount := std.Coins{std.NewCoin(ugnot.Denom, 10e8)}
+	balances := []string{
+		fmt.Sprintf("%s=%s", myVal.GetPubKey().Address().String(), defaultBalanceAmount.String()),
+		fmt.Sprintf("%s=%s", dKeys[0].Address().String(), defaultBalanceAmount.String()),
+		fmt.Sprintf("%s=%s", dKeys[1].Address().String(), defaultBalanceAmount.String()),
+		fmt.Sprintf("%s=%s", dKeys[2].Address().String(), defaultBalanceAmount.String()),
+	}
+
+	balanceSheet := filepath.Join(t.TempDir(), "balance-sheet.txt")
+	err = os.WriteFile(balanceSheet, []byte(strings.Join(balances, "\n")), 0644)
+	require.NoError(t, err)
+
+	// Add balance sheet
+	runGnoGenesisCommand(t, ctx, "balances", "add",
+		"--genesis-path", genesisfile,
+		"--balance-sheet", balanceSheet,
+	)
+
+	// Set some whitelisted addresses
+	runGnoGenesisCommand(t, ctx, "params", "set",
+		"--genesis-path", genesisfile,
+		"auth.unrestricted_addrs",
+		dKeys[0].Address().String(),
+		dKeys[1].Address().String(),
+		dKeys[2].Address().String(),
+	)
+
+	// Set restricted denom
+	runGnoGenesisCommand(t, ctx, "params", "set",
+		"--genesis-path", genesisfile,
+		"bank.restricted_denoms", "ugnot",
+	)
+
+	// XXX(gfanton): add some validators, they should load a genesis as well and we
```

Comment:
We're gonna have our testing harness (`tessera`) do this, so no worries on having this in a unit test right now